src/main/scala/com/example/reporters/common/DfLogReporter.scala

```diff
@@ -0,0 +1,11 @@
+package com.example.reporters.common
+
+import org.apache.spark.sql.SaveMode
+
+trait DfLogReporter extends Reporter {
+  val reportData: ReportDfUnit
```

Comment:
Could you explain the decision with reportData (this holds the result, right?) within case classes for reporters.
Not sure if this is good functional approach. 
Please elaborate on the benefits of this option, also it would be great if you find some alternatives and compare.

Reply:
The encupsulation Case classes was choosen due to immutability. As for the Reporter trait it helps to encapsulate reporing logic. I was searching for patterns/common practices of spark reporting, but without effect.

Reply:
this guy writes a lot of interesting articles about better organizing code in spark https://medium.com/@mrpowers
E.g. this one is helpful: https://medium.com/@mrpowers/how-to-write-spark-etl-processes-df01b0c1bec9